woocommerce/woocommerce-gateway-stripe
# Pass customer language/locale to Stripe upon creation or modification.

# Changes proposed in this Pull Request:

Issue: 
Fixes #744

Description: 
When a customer is either created or modified the `generate_customer_request` method is called. Within that, I added a call to the new method `get_customer_preferred_locale` that gets the customer locale based on the user or site settings. If the user/site locale does not match [one of the supported Stripe locales](https://support.stripe.com/questions/language-options-for-customer-emails), an empty array is returned.


# Testing instructions

Testing instructions: How should this be tested and how can a reviewer test the end-user functionality? Are there known issues that you plan to address in a future PR? Are there any side effects that readers should be aware of?

1. Set your user or site language to [one of the supported Stripe languages](https://support.stripe.com/questions/language-options-for-customer-emails)
2. Make a purchase.
3. Check the purchase, and then the customer in Stripe.
4. The customer's Language under Details should be the language set for your user/site. 

![Markup 2021-03-31 at 13 30 52](https://user-images.githubusercontent.com/4634416/113178906-6cb24600-9225-11eb-9272-07cdcea03515.png)

# Notes
I was going to make tests for this, but since this relies on functions and not methods it was difficult to attempt to mock without getting complicated. An alternate testing route would have been to set up `get_customer_locale` to return set values, but that would basically be handing the expected return to the test. If there are any ideas on this, I am willing to work on them.
